network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/ConnectablePositionImpl.java

```diff
@@ -68,9 +103,105 @@ public Direction getDirection() {
 
         @Override
         public Feeder setDirection(Direction direction) {
-            getAttributes().setDirection(Direction.valueOf(Objects.requireNonNull(direction).name()));
+            Direction oldValue = getAttributes().getDirection();
+            if (!Objects.equals(oldValue, direction)) {
+                if (getExtendable() instanceof AbstractInjectionImpl<?, ?> injection) {
+                    updateInjectionDirectionResource(injection, direction);
+                    notifyUpdate(injection, DIRECTION, oldValue, direction);
+                } else if (getExtendable() instanceof AbstractBranchImpl<?, ?> branch) {
+                    updateBranchDirectionResource(branch, direction);
+                    notifyUpdate(branch, DIRECTION, oldValue, direction);
+                } else if (getExtendable() instanceof ThreeWindingsTransformerImpl windingsTransformer) {
+                    updateTransformerDirectionResource(windingsTransformer, direction);
+                    notifyUpdate(windingsTransformer, DIRECTION, oldValue, direction);
+                }
+            }
             return this;
         }
+
+        private void updateInjectionLabelResource(AbstractInjectionImpl<?, ?> injection, String name) {
+            injection.updateResource(res -> res.getAttributes().getPosition().setLabel(Objects.requireNonNull(name)));
+        }
+
+        private void updateBranchLabelResource(AbstractBranchImpl<?, ?> branch, String name) {
+            branch.updateResource(res -> {
+                if (getSide() == ThreeSides.ONE) {
+                    res.getAttributes().getPosition1().setLabel(Objects.requireNonNull(name));
+                } else if (getSide() == ThreeSides.TWO) {
+                    res.getAttributes().getPosition2().setLabel(Objects.requireNonNull(name));
+                }
+            });
+        }
+
+        private void updateTransformerLabelResource(ThreeWindingsTransformerImpl windingsTransformer, String name) {
+            windingsTransformer.updateResource(res -> {
+                if (getSide() == ThreeSides.ONE) {
+                    res.getAttributes().getPosition1().setLabel(Objects.requireNonNull(name));
+                } else if (getSide() == ThreeSides.TWO) {
+                    res.getAttributes().getPosition2().setLabel(Objects.requireNonNull(name));
+                } else if (getSide() == ThreeSides.THREE) {
```

Comment:
maybe should just be else ?

Reply:
an else to check and handle the case where the side is not supported